### PR TITLE
feat(medusa): Improve add line item to cart perf and transaction management + clustering start command

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Improve the transaction management form add to cart end point
- add a new cluster command to start the server. By default, it will consume all available cpus to start the medusa instances. An option is available to specify the number of CPU to use

### CODE REVIEW

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts:72]: can we await with `promiseAll()` line 147 and 157 to be completed?

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

I just left 1 suggestion we can try

File: packages/medusa/src/strategies/cart-completion.ts
Comment (pKorsholm) [inline packages/medusa/src/strategies/cart-completion.ts]:
```suggestion
    let cart = await txCartService.retrieve(id, {
```

**Review by pKorsholm — APPROVED** [review]
LGTM! just a minor nit 😄

**Comment by adrien2p** [comment]
Before merging I am just looking at another point

**Comment by olivermrbl** [comment]
@adrien2p ~~do you think we should include this in the Recap release?~~

I think we should exclude this from the upcoming release, which has already been through testing. So please do not merge before 1.18.1 is out :)

**Comment by adrien2p** [comment]
> @adrien2p do you think we should include this in the Recap release?

I think we could, especially the cluster start command wdyt?

 I am also looking at some points in the create cart workflow cause there is currently multiple duplication which could be prevented. But I am just doing some tests and it would eventually be a different pr

**Comment by adrien2p** [comment]
@olivermrbl should we also stress the user that using the cluster in product is best practice? [see](https://expressjs.com/en/advanced/best-practice-performance.html#run-your-app-in-a-cluster)

File: packages/medusa-cli/src/create-cli.ts
Comment (olivermrbl) [inline packages/medusa-cli/src/create-cli.ts:236]: comment: needs documentation

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts]: thought: Should we separate this into two functions? I imagine this will live as two steps in the workflow in the future, so it would make for an easier migration.

Comment (olivermrbl) [reply packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts]: I realise we are "collecting" promises here, which is good for performance, so I am also fine with keeping it as is

File: packages/medusa/src/strategies/cart-completion.ts
Comment (olivermrbl) [inline packages/medusa/src/strategies/cart-completion.ts:249]: question: Why are we converting `authorizePayment` to accept a cart, if we are not going to use it?

**Review by olivermrbl — APPROVED** [review]
LGTM, with comments. Great work :)

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts
Comment (adrien2p) [reply packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts]: done

File: packages/medusa/src/strategies/cart-completion.ts
Comment (adrien2p) [reply packages/medusa/src/strategies/cart-completion.ts:249]: reverted

File: packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts]: question: Shouldn't we delete `handleAddOrUpdateLineItem`? It is no longer used as far as I can see

**Pull request merged**
